core/src/main/java/org/elasticsearch/action/bulk/BulkItemRequest.java

```diff
@@ -94,15 +94,36 @@ public void readFrom(StreamInput in) throws IOException {
         request = DocWriteRequest.readDocumentRequest(in);
         if (in.readBoolean()) {
             primaryResponse = BulkItemResponse.readBulkItem(in);
+            if (in.getVersion().onOrAfter(Version.V_5_6_0_UNRELEASED)) {
```

Comment:
We discussed this via another channel, @bleskes will work on removing this guard and obviating the need for even considering adding a 6.0.0 version constant.